File: app/controllers/EvaluationController.java
package controllers;

import controllers.core.BaseController;
import model.Secured;
import model.feedback.EvaluatedAspect;
import model.feedback.Feedback;
import model.feedback.Feedback.EvaluatedTool;
import model.feedback.FeedbackKey;
import model.feedback.Mark;
import model.user.User;
import play.data.DynamicForm;
import play.data.FormFactory;
import play.mvc.Result;
import play.mvc.Security.Authenticated;

import javax.inject.Inject;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

@Authenticated(Secured.class) public class EvaluationController extends BaseController {

  @Inject public EvaluationController(FormFactory f) {
    super(f);
  }

  public Result index() {
    User user = getUser();
    List<Feedback> toEvaluate = Arrays.stream(Feedback.EvaluatedTool.values()).map(tool -> {
      FeedbackKey key = new FeedbackKey(user.name, tool);
      Feedback feedback = Feedback.finder.byId(key);
      if(feedback == null)
        feedback = new Feedback(key);
      return feedback;
    }).collect(Collectors.toList());

    return ok(views.html.evaluation.eval.render(user, toEvaluate));
  }

  public Result submit() {
    User user = getUser();

    DynamicForm form = factory().form().bindFromRequest();

    List<Feedback> evaluation = Arrays.stream(EvaluatedTool.values()).map(tool -> readFeedback(user, form, tool))
        .collect(Collectors.toList());

    for(Feedback f : evaluation)
      f.save();

    return ok(views.html.evaluation.submit.render(user, evaluation));
  }

  private Feedback readFeedback(User user, DynamicForm form, EvaluatedTool tool) {
    FeedbackKey key = new FeedbackKey(user.name, tool);
    Feedback feedback = Feedback.finder.byId(key);
    if(feedback == null)
      feedback = new Feedback(key);

    String evaluatedTool = tool.toString().toLowerCase();

    for(EvaluatedAspect evaledAspect : EvaluatedAspect.values()) {
      feedback.set(evaledAspect, Mark.valueOf(
          form.get(evaledAspect.toString().toLowerCase() + "-" + evaluatedTool.toLowerCase()).toUpperCase()));
    }

    feedback.comment = form.get("comment-" + evaluatedTool);

    return feedback;
  }

}

File: app/controllers/EvaluationController.scala
package controllers

import javax.inject._

import model.Enums.EvaluatedAspect
import model.User
import model.core.Repository
import model.feedback._
import play.api.db.slick.{DatabaseConfigProvider, HasDatabaseConfigProvider}
import play.api.mvc._
import slick.jdbc.JdbcProfile

import scala.concurrent.ExecutionContext

class EvaluationController @Inject()(cc: ControllerComponents, val dbConfigProvider: DatabaseConfigProvider, val repository: Repository)
                                    (implicit ec: ExecutionContext)
  extends AbstractController(cc) with HasDatabaseConfigProvider[JdbcProfile] with Secured {

  def index: EssentialAction = withUser { user =>
    implicit request =>
      //      val toEvaluate: Seq[Feedback] = Feedback.EvaluatedTool.values.map(tool => {
      //        val key = new FeedbackKey(user.name, tool)
      //        Option(Feedback.finder.byId(key)).getOrElse(new Feedback(key))
      //      }).toList
      val toEvaluate: Seq[Feedback] = Seq.empty
      Ok(views.html.evaluation.eval(user, toEvaluate))
  }

  def submit: EssentialAction = withUser { user =>
    implicit request =>
      val evaluation: Seq[Feedback] = EvaluatedTool.values map (tool => readFeedback(user, tool))

      //      evaluation.foreach(_.save)

      Ok(views.html.evaluation.submit(user, evaluation))
  }

  def readFeedback(user: User, tool: EvaluatedTool)(implicit request: Request[AnyContent]): Feedback = {
    //    val key = new FeedbackKey(user.name, tool)
    //    val feedback = Option(Feedback.finder.byId(key)).getOrElse(new Feedback(key))

    val evaluatedTool = tool.toString.toLowerCase

    for (evaledAspect <- EvaluatedAspect.values) {
      //      val markStr = singleStrForm(evaledAspect.toString.toLowerCase + "-" + evaluatedTool.toLowerCase).bindFromRequest.get.str
      //      feedback.set(evaledAspect, Mark.valueOf(markStr.toUpperCase))
    }

    //    feedback.comment = singleStrForm("comment-" + evaluatedTool).bindFromRequest.get.str

    //    feedback
    null
  }

}
